ext/oj-introspect/introspect.c

```diff
@@ -36,6 +42,31 @@ static void start(ojParser p) {
     d->delegated_start_func(p);
     // Reset to zero so the parser and delegate can be reused.
     d->byte_offsets.current = 0;
+
+    /*
+    * If the `filter` is provided we will start introspecting later
+    * once we encounter with the key provided.
+    */
+    d->introspect = (d->filter == NULL);
+}
+
+static void copy_ruby_str(char **target, VALUE source) {
+  size_t len = RSTRING_LEN(source);
+  *target = ALLOC_N(char, len + 1);
+  memcpy(*target, RSTRING_PTR(source), len);
+  (*target)[len] = '\0'; // Parantheses are important as it means => (*target + sizeof(char) * len)
```

Comment:
The RSTRING_PTR should return a \0 terminated string but this is still fine.

Reply:
Ruby extension doc is saying that it may not be nul-terminated 🤔 But as you mentioned, it should be fine in either case.

Thanks a lot for your review 🙏